src/ripple/consensus/Validations.h

```diff
@@ -580,17 +584,17 @@ class Validations
             // Check that validation sequence is greater than any non-expired
             // validations sequence from that validator
             auto const now = byLedger_.clock().now();
-            SeqEnforcer<Seq>& enforcer = seqEnforcers_[key];
+            SeqEnforcer<Seq>& enforcer = seqEnforcers_[nodeID];
             if (!enforcer(now, val.seq(), parms_))
                 return ValStatus::badSeq;
 
             // This validation is a repeat if we already have
-            // one with the same id for this key
-            auto const ret = byLedger_[val.ledgerID()].emplace(key, val);
-            if (!ret.second && ret.first->second.key() == val.key())
-                return ValStatus::repeatID;
+            // one with the same id and signing key for this node
+            auto ret = byLedger_[val.ledgerID()].emplace(nodeID, val);
+            if (!ret.second)
+                ret.first->second = val;
```

Comment:
I'd usually see code like this written as something like: `byLedger_[ledgerID][nodeID] = val`. What you have is better (and doesn't require val to be default constructable). Maybe leave a comment that we should use c++-17's `insert_or_assign` here when we move to c++-17. This both clarifies the intent here and reminds us to clean this up when we switch.